cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -287,7 +287,14 @@ impl CoreEnvironment<ReadOnly> {
         contents: String,
     ) -> Result<EditResult, CoreEnvironmentError> {
         let old_contents = self.manifest_content()?;
-        // TODO we should probably skip this if the manifest hasn't changed
+
+        // skip the edit if the contents are unchanged
+        // note: consumers of this function may call [Self::link] separately,
+        //       causing an evaluation/build of the environment.
+        if contents == old_contents {
```

Comment:
Suggestion: We can probably be a little bit more thorough about what is considered a change. Right now this will rebuild even if you edit a comment. You could probably cover the large majority of changes by just parsing to TOML (via `toml`, not `toml_edit`, that doesn't implement `PartialEq`) and checking the equality of that struct.

Reply:
We still need to propagate the manifest into the lock, even for a whitespace change.

Reply:
the reason i didnt want to go with toml contents is that
a) graham asked for optimizing the "quitting the editor without a change" case
b) we probably do want new generations when only the comments are changed in case of managed/remote envs. Since this is implemented on the core env level returning unchanged in case of trivia changes would prevent that.